loopin.go

```diff
@@ -840,6 +890,18 @@ func (s *loopInSwap) waitForSwapComplete(ctx context.Context,
 	htlcKeyRevealed := false
 	for !htlcSpend || !invoiceFinalized {
 		select {
+		// If the client requested the swap to be abandoned, we override
+		// the status in the database.
+		case <-s.abandonChan:
+			err = s.abandonSwapCancelInvoice(ctx)
+			if err != nil {
+				return err
+			}
+
+			return fmt.Errorf("swap hash "+
```

Comment:
We rely on the error here, it might be that `abandonSwapCancelInvoice()` is not returning an error. Also the function might fail due db issues which we want to catch here.